### PR TITLE
Fix/1969 family table

### PR DESCRIPTION
**Pull request by Velythyl**
The family table now takes the new family data as input. No more need to query GraphQL for every family member.

### CODE REVIEW

File: src/components/EntityPage/Participant/Utils/FamilyTable.js [hunk 1]
```diff
@@ -14,15 +11,56 @@ class FamilyTable extends React.Component {
   constructor(props) {
     super(props);
 
-    this.state = { ready: false };
-
     const participant = props.participant;
+    participant.relationsip = "this participant";
```
Comment (nyanofthemoon) [inline src/components/EntityPage/Participant/Utils/FamilyTable.js]: Sip relations? :D

File: src/components/EntityPage/Participant/Utils/FamilyTable.js [hunk 2]
```diff
@@ -47,168 +85,82 @@ class FamilyTable extends React.Component {
...
+      baseline('Vital Status', 'outcome.vital_status'),
+      baseline('Gender', 'gender'),
+      baseline('Diagnoses (Mondo)', 'diagnoses.hits.edges', true, 'mondo_id_diagnosis'),
+      baseline('Diagnoses (NCIT)', 'diagnoses.hits.edges', true, 'ncit_id_diagnosis'),
+    ];
+
+    return famMembersNodes.reduce( (rows, node) => {  //reduce the family members into rows of a table
+      const kf_id = node.kf_id;
+
+      return rows.flatMap( currentRow => {  //map the rows into more rows, splicing in new rows as needed with flatMap's unpacking
```
Comment (nyanofthemoon) [inline src/components/EntityPage/Participant/Utils/FamilyTable.js]: IE support pretty plz?

File: src/components/EntityPage/Participant/Utils/FamilyTable.js [hunk 3]
```diff
@@ -47,168 +80,82 @@ class FamilyTable extends React.Component {
...
+
+        if(Array.isArray(accessorItem)) { //if the item is an array, then we'll have to extract some subaccessors from the array items
+
+          //we return an array when we want to splice our values at this index: since we're using flatMap, it'll unpack them in the right positions for us!
+          //if the value we want to splice in is an empty array, no biggie, it will be unpacked into, well, nothing
+          return [currentRow].concat(accessorItem.map(a => get(a.node, currentRow.subacc, null)).reduce( (acc, item) => {   //reduce the accessed array into new rows
+            let subRow = rows.find( (ele) => ele.leftField === item); //let's try to see if the value is already in there.
+
+            if(subRow) subRow[kf_id] = "reported";  //if it is, great, let's just mutate it.
+            else {
```
Comment (nyanofthemoon) [inline src/components/EntityPage/Participant/Utils/FamilyTable.js]: y manques-tu le cote gauche du else?

File: src/components/EntityPage/Participant/Utils/FamilyTable.js (hunk 1 above)
Comment (Velythyl) [reply src/components/EntityPage/Participant/Utils/FamilyTable.js]: Done ;_;

File: src/components/EntityPage/Participant/Utils/FamilyTable.js (hunk 2 above)
Comment (Velythyl) [reply src/components/EntityPage/Participant/Utils/FamilyTable.js]: Done with lodash!

**Pull request merged**